### DISCUSSION TITLE
Invalid next.config.js options detected after upgrade to 12.2.5

### DISCUSSION THREAD

**Discussion by Mar-he** [vercel/next.js#39892] [Help]
With the following next-18next.config.js:

```js
const I18NextHttpBackend = require('i18next-http-backend/cjs')

module.exports = {
    i18n: {
        defaultLocale: 'de',
        locales: ['de', 'en'],
        backend: {
            loadPath: `${process.env.INTERNAL_API_URI}/api/locales/{{lng}}/{{ns}}`
        },
    },
    debug: false,
    reloadOnPrerender: true,
    ns: ["common", "employees", "projects", "userprofile"],
    serializeConfig: false,
    use: [I18NextHttpBackend]
}
```

and this next.config.js

```js
const { i18n } = require('./next-i18next.config')
const withLess = require("next-with-less");

const path = require('path')
const basePath = process.env.BASE_PATH || ``;

module.exports = withLess({
  i18n,
  basePath: basePath,

  lessLoaderOptions: {
    additionalData: (content) =>
      `${content}\n\n@import '${path.resolve(__dirname, "./styles/theme.less")}';`,
  },
});
```
 I am receiving the following warning, after uprading from `^12.0.10` to `^12.2.5`:

warn  - Invalid next.config.js options detected:
  - The value at .i18n has an unexpected property, backend, which is not in the list of allowed properties (defaultLocale, domains, localeDetection, locales).

Why is this happening and how can I resolve the issue?

**Comment by icyJoseph** [comment]
The type definitions, although not the source of config warning, do not show a backend key in the i18n object:

https://github.com/vercel/next.js/blob/canary/packages/next/server/config-shared.ts#L20

I am not sure about the implications of this warning though. Perhaps is worth reading up on i18next-http-backend, I think, besides the warning, it might just work.

**Comment by Ali-Hussein-dev** [comment]
@icyJoseph 
I have similar issue with next config after 12.2.5. 

```js
const withPlugins = require('next-compose-plugins');
const withNx = require('@nrwl/next/plugins/with-nx');
const withPWA = require('next-pwa');

// @required for mono-repo NEXT_TRANSLATE_PATH
// https://github.com/vinissimus/next-translate#13-how-to-use-next-translate-in-a-mono-repo
const nextTranslate = require('next-translate');
/**
 * @type {import('@nrwl/next/plugins/with-nx').WithNxOptions}
 **/
const isDev = process.env.NEXT_PUBLIC_VERCEL_ENV === 'development';

const customConfig = {
  pwa: {
    dest: '../../dist/apps/kartebuch/public',
    disable: isDev,
  },
};
module.exports = withPlugins([withNx, nextTranslate, withPWA], customConfig);
```

`Invalid next.config.js options detected: `
`  - The root value has an unexpected property, webpackDevMiddleware, which is not in the list of allowed properties (amp,...).`
`- The root value has an unexpected property, target, which is not in the list of allowed properties (amp,...)`
